src/coord/src/catalog/storage.rs

```diff
@@ -528,24 +555,30 @@ impl Transaction<'_> {
         timestamp: Timestamp,
         offset: i64,
     ) -> Result<(), Error> {
-        match self
+        let result = self
             .inner
             .prepare_cached(
                 "INSERT OR IGNORE INTO timestamps (sid, pid, timestamp, offset) VALUES (?, ?, ?, ?)",
             )?
-            .execute(params![SqlVal(source_id), partition_id, timestamp, offset])
-        {
+              .execute(params![SqlVal(source_id), partition_id, timestamp, offset]);
+
+        self.assert_timestamp_bindings(source_id);
+
+        match result {
             Ok(_) => Ok(()),
             Err(err) => Err(err.into()),
         }
     }
 
     pub fn delete_timestamp_bindings(&self, source_id: GlobalId) -> Result<(), Error> {
-        match self
+        let result = self
             .inner
             .prepare_cached("DELETE FROM timestamps WHERE sid = ?")?
-            .execute(params![SqlVal(&source_id)])
-        {
+            .execute(params![SqlVal(&source_id)]);
+
+        self.assert_timestamp_bindings(&source_id);
```

Comment:
Do you think we might plan to add more assertions in the future? With the one currently in this PR, this assert could never fire after a delete